tests/testthat/test-log-likelihood-logistic.R

```diff
@@ -28,3 +29,48 @@ test_that("logistic.log.likelihood.gradient against numerical gradient",
               )
             )
           })
+
+
+test_that("take.one.newton.step QR option agrees with LU ", {
+  n_obs <- 32
+  n_pred <- 4
+  data <-
+    simulate_data(n_obs, n_pred, model = 'logit', seed = 1928)
+  design <- data$design
+  outcome <- data$outcome
+  start_c <- rep(0, n_pred)
+
+  lu_out <- take.one.newton.step(design, outcome, start_c, "LU")
```

Comment:
you can test at an even lower level here, since take.one.newton.step just sends the arguments to the appropriate solver. So testing rcpp_qr vs qr.solve. Ifyou wanted to be thorough, the test for take.one.newton.step would test whether the right function was called.